container-images/tcib/base/os/tempest/run_tempest.sh

```diff
@@ -423,6 +423,14 @@ function generate_test_results {
     TEMPEST_LOGS_DIR=${TEMPEST_PATH}${TEMPEST_WORKFLOW_STEP_DIR_NAME}/
     mkdir -p ${TEMPEST_LOGS_DIR}
 
+    echo "Failing tests marked as expected failures"
+    if [ ! -z ${TEMPEST_EXPECTED_FAILURES_LIST} ]; then
+        stestr failing --list | sed 's/\[.*\]//g' > ${TEMPEST_LOGS_DIR}stestr_failing.txt
+        if grep -Fxf ${TEMPEST_LOGS_DIR}stestr_failing.txt ${TEMPEST_EXPECTED_FAILURES_LIST}; then
+            RETURN_VALUE=0
+        fi
+    fi
+
```

Comment:
The reason I put it here is because it creates output similar to include and exclude list and also I take advantage of the variable `TEMPEST_LOGS_DIR` to save failing tests file. Or do you think there is a better place to store the file? Please let me know :smile:

Reply:
The name of the function is `generate_test_results`. Therefore it should do just generating of logs. If you want to keep the `stestr_failing.txt` in the collected logs then I suggest you generate the file inside of this function:

```
stestr failing --list | sed 's/\[.*\]//g' > ${TEMPEST_LOGS_DIR}stestr_failing.txt
```

Then the rest of the code (changing of the `RETURN_VALUE`) can be moved to the end of the file [#489](https://github.com/openstack-k8s-operators/tcib/pull/226/files/449c7ebf055959573d7838d2e97e79442bb69958#diff-191bc7762340f2539b035115667e34a49916229d47d776aca4a2c457f694e35cR489)

```
if [ -f ${TEMPEST_LOGS_DIR}stestr_failing.txt ]; then
  grep -Fxf ${TEMPEST_LOGS_DIR}stestr_failing.txt ${TEMPEST_EXPECTED_FAILURES_LIST} && RETURN_VALUE=0
fi
```

You can move the `TEMPEST_LOGS_DIR` to line [#99](https://github.com/openstack-k8s-operators/tcib/pull/226/files/449c7ebf055959573d7838d2e97e79442bb69958#diff-191bc7762340f2539b035115667e34a49916229d47d776aca4a2c457f694e35cR99). Wdyt?:)

Reply:
I have made changes similar to what you meant. What do you think about this version? :D